doc/developer/feature-branch.rst

```diff
@@ -0,0 +1,109 @@
+only which will simplify the process, but close the door to not mainteners)*
+could ask for a new feature branch by following the bellow process:
+
+- **Prerequisit:** If feature is not listed in FRR Wiki Features Requests,
+  first create a new issue in FRR GitHub to request the new feature and wait
+  the feature is added to the Wiki
+- **Request:** [Send a simple mail is send to @frr-dev mailing list | Fill a
+  new issue | Update issue of the feature] *(NOTE: we need to choose the
+  appropriate method)* with a short description of the devlopment with the
+  issue number *(if mail method is choosen)*, and, if known, the list of
+  partners that express to work on. *(NOTE: these are the minimum information
+  I thought of. Of course additional information could be added)*
+- **Examination**: Request is disscussed during [the regular Tuesday meeting |
+  the TSC meeting] *(NOTE: we need to choose between the 2 options)*.
+- **Decision**: If request is approved, corresponding branch is created (see
+  below). A mail is sent to the @frr-dev mailing list to announce the creation
+  of the new branch and a link to that branch is added to the issue and the
+  FRR Wiki Features Requests page. If it is rejected, a mail with the reasons
+  of the rejection is also sent to the @frr-dev mailing list *(NOTE: if GitHub
+  issue method is choosen for the request, above text must be adapted, but,
+  I think that we could keep the announcement throuhg the mailing list)*.
+- **Creation**: If the request is approved, branch **dev-XXX** is created by
+  a TSC member or any personn that have the right to do it. A Owner of the
+  feature branch is designated as the responsible of this new feature branch.
+  The Owner is [a maintainers designated to shepherd the development on this
+  feature branch | the requester of the feature branch who is promoted as
+  maintainer to get commit rights on this branch] *(NOTE: we must decide if
+  we go for a shepherded maintainer or promote the requester if (s)he is not
+  already)*.
+
+
+Feature Branch Life Cycle
+-------------------------
+
+Once created the Feature Branch is used as follow:
+
+- Pull Requests are tagged with the corresponding “dev label” and assigned to
+  the owner of the feature branch or the shepherded maintainer
+- Optional: a light review is performed by contributors interested by this 
+  development
+- [Owner of the feature branch | Shepherded maintainers] *(NOTE: depends of
+  what has been choosen previously)* commit the Pull Request
+- Owner of the feature branch, if (s)he as the rights (i.e. the Owner is a
+  maintainer or (s)he has been promoted - see above) could directly commit code
+
+Once development is finished, README.md is replaced by the version from master
+branch and a final rebase againts master (eventually after squashing, commit
+cleaning, commit reordering …) is performed in order to submit a Pull Request
```

Comment:
when you talk about rebase, that means you accept to overwrite the history of dev-XXX branch ?
I think if needed, you should keep the history, and instantiate a new branch: eg: dev-xxx branch for the initial one, then dev-xxx-20192910 for a rebase to master.
somehow, the old dev-xxx branch should be deprecated or "closed"